src/transformers/models/jukebox/modeling_jukebox.py

```diff
@@ -0,0 +1,2974 @@
+        mask=False,
+        zero_out=False,
+        init_scale=1.0,
+        res_scale=False,
+        m_attn=0.25,
+        m_mlp=1.0,
+        attn_order=0,
+        blocks=None,
+        spread=None,
+        encoder_dims=None,
+        lyric_enc_len=None,
+    ):
+        super().__init__()
+        self.width = width
+        self.n_ctx = n_ctx
+        self.encoder_dims = encoder_dims
+        self.blocks = blocks
+        if blocks is not None:
+            self.block_ctx = n_ctx // blocks
+        self.lyric_enc_len = lyric_enc_len
+        self.num_heads = num_heads
+
+        res_scale = 1.0 / n_depth if res_scale else 1.0
+
+        # Orders of attn_func
+        attn_func = self.get_attn_func(attn_order)
+
+        def attn_block(depth):
+            return JukeboxBlock(
+                width=width,
+                n_ctx=n_ctx,
+                num_heads=num_heads,
+                attn_dropout=attn_dropout,
+                resid_dropout=resid_dropout,
+                afn=afn,
+                scale=scale,
+                mask=mask,
+                zero_out=zero_out if attn_func(depth) != 6 else True,
+                init_scale=init_scale,
+                res_scale=res_scale,
+                m_attn=m_attn,
+                m_mlp=m_mlp,
+                attn_func=attn_func(depth),
+                blocks=blocks,
+                spread=spread,
+                encoder_dims=encoder_dims,
+                lyric_enc_len=lyric_enc_len,
+            )
+
+        self._attn_mods = nn.ModuleList()
+        for depth in range(n_depth):
+            self._attn_mods.append(attn_block(depth))
+
+        self.saved_attn_weights = []
+
+    def get_attn_func(self, attn_order: int):
+        """
+        Get the correct attention order pattern.
+        """
+        mapping = {
+            0: lambda layer: 0,
+            1: lambda layer: [1, 2][layer % 2],
+            2: lambda layer: [1, 2, 3][layer % 3],  # Alternate row, column and previous row attn
+            3: lambda layer: [1, 4][layer % 2],  # Alternate row and last column
+            4: lambda layer: [1, 5][layer % 2],  # Alternate row and last k columns
+            5: lambda layer: [1, 4, 1, 1][layer % 4],  # Alternate row, last column, row, row
+            6: lambda layer: [1, 2, 3, 6][layer % 4],
+            7: lambda layer: [*[1, 2, 3] * 5, 6][layer % 16],
+            8: lambda layer: [1, 2, 3, 1, 2, 3, 1, 2, 3, 6][layer % 10],  # Used by separated_enc_dec model with lyrics
+            9: lambda layer: [1, 2, 3, 0][layer % 4],
+            # Used by large separated_enc_dec model with lyrics
+            10: lambda layer: [*[1, 2, 3, 1, 2, 3, 1, 2, 3], *[1, 2, 3, 1, 2, 3, 1, 2, 3, 6] * 7][layer % 79],
+            11: lambda layer: [6, 6, 0][layer % 3] if layer % 16 == 15 else [1, 2, 3][layer % 3],
+            # Used by single_enc_dec model with lyrics
+            12: lambda layer: [7, 7, 0][layer % 3] if layer % 16 == 15 else [1, 2, 3][layer % 3],
+        }
```

Comment:
This kind of code is unmaintainable, we need something different here. Again an nice enum would be more readable.